aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -94,6 +94,27 @@ SparseTensor& mul_sparse_scalar_(SparseTensor& t, Scalar v) {
   return mul_out_sparse_scalar(t, t, v);
 }
 
+// --------------------------------------------------------------------
+// log1p(SparseTensor)
+// --------------------------------------------------------------------
+
+SparseTensor& log1p_out_sparse(SparseTensor& r, const SparseTensor& t) {
+  AT_ASSERT(r.is_sparse());
+  AT_ASSERT(t.is_sparse());
+
+  if (isSameTensor(r, t)) {
+    r._values().log1p_();
```

Comment:
Is this valid if r is not coalesced? Same with below.

Reply:
oh, it is not... I will remove the inplace ops all together..

Reply:
I don't think you need to remove the inplace ops, can we just coalesce before doing this op?

Reply:
Yeah, you are right. I will coalesce first

Reply:
oh, wait, sounds like we don't have an inplace coalesce right now.  Maybe we might not want to support the inplace log1p for now?

Reply:
Okay yeah, let's forget about the inplace version for now.

Reply:
By the way, we should have an inplace coalesce. One for the list.